### DISCUSSION TITLE
HLS caching

### DISCUSSION THREAD

**Discussion by lnxct** [TheWidlarzGroup/react-native-video#4950] [Ideas]
### What problem are you trying to solve?

It continues to hit the CDN URL even after playing one time. So it is increased cost, use of user's data continues.

### Proposed solution

HLS caching. HLS caching allow to the storage of .ts and m3u8 file.

**Comment by flexgrip** [comment]
I'm unsure what RNV would have to do with this. Caching hls would be pretty tough on storage requirements. And it's not a good format for storage. Just for streaming.

Also, iOS natively caches chunks for quicker timeline scrubbing.

But no. I don't think rnv should be responsible for this. I have an app that lets the user stream a video, or download it for later. If I made it automatically store the videos, they'd fill up their storage quick.

**Comment by evoactivity** [comment]
@flexgrip just so you are aware, caching is something that is being worked on for android in this branch https://github.com/react-native-video/react-native-video/pull/2708

I don't think the idea is to store everything forever, rather they stick around for the session and the cache size is limited

**Comment by nguyenvanphituoc** [comment]
hello, I have implement hls cache, that suitable with react-native-video: https://github.com/nguyenvanphituoc/react-native-cache-video#readme

**Comment by IslamRustamov** [comment]
Here is the PR into this repo that enables it https://github.com/TheWidlarzGroup/react-native-video/pull/4272 for android

**Comment by TWG-AI** [comment]
Thanks for the idea — noted for the team to consider.

> I'm an AI assistant, so double-check any specifics I may have mentioned.

**Comment by TWG-AI** [comment]
Quick update: Android HLS segment caching was added in [PR #4272](https://github.com/TheWidlarzGroup/react-native-video/pull/4272) (merged). iOS HLS caching is still open — that's the remaining scope for this request.

> I'm an AI assistant — verify the Android implementation details in the PR.
